src/boot/efi/meson.build

```diff
@@ -266,11 +266,19 @@ efi_ldflags = [
         efi_crt0,
 ]
 
-possible_link_flags = [
-        '-Wl,--no-warn-execstack',
-        '-Wl,--no-warn-rwx-segments',
-]
-efi_ldflags += cc.get_supported_link_arguments(possible_link_flags)
+foreach arg : ['-Wl,--no-warn-execstack',
+               '-Wl,--no-warn-rwx-segments']
+        # We need to check the correct linker for supported args. This is what
+        # cc.has_multi_link_arguments() is for, but it helpfully overrides our
+        # choice of linker by putting its own -fuse-ld= arg after ours.
+        if run_command('bash', '-c',
+                       'exec $1 -x c -o/dev/null <(echo "int main(void){return 0;}")' +
```

Comment:
Before I test I have two doubts regarding `exec $1` part:
1. `bash` man page states:
    ```
       -c        If the -c option is present, then commands are read from the first non-option argument command_string.  If
                 there are arguments after the command_string, the first argument is assigned to $0 and any remaining argu‐
                 ments  are  assigned to the positional parameters.  The assignment to $0 sets the name of the shell, which
                 is used in warning and error messages.
    ```
    so if ie `CC=gcc` isn't `gcc` end up as `$0` and `$1` will be empty?
2. What happens if ie `CC="ccache gcc"`?

Reply:
Dunno why I wasn't using `$@`. I got lucky that meson was using `ccache cc`.